.github/workflows/ci-lint.yml

```diff
@@ -0,0 +1,54 @@
+name: CI
```

Comment:
Decoupling the running of our lint tools (which can be run in any node version) from the running of our tests (which should only be run against the versions of node we support in `package.json#engines`). See https://github.com/Shopify/polaris-internal/issues/1522 for the differences.